test/functional/wallet_basic.py

```diff
@@ -338,6 +373,52 @@ def run_test(self):
         # This will raise an exception for importing an invalid pubkey
         assert_raises_rpc_error(-5, "Pubkey is not a valid public key", self.nodes[0].importpubkey, "5361746f736869204e616b616d6f746f")
 
+        # send with explicit btc/kb fee
+        self.sync_all(self.nodes[0:3])
+        self.log.info("test explicit fee (sendtoaddress as btc/kb)")
+        self.nodes[0].generate(1)
+        prebalance = self.nodes[2].getbalance()
+        assert prebalance > 2
+        address = self.nodes[1].getnewaddress()
+        # Throw if no conf_target provided
+        assert_raises_rpc_error(-8, "Selected estimate_mode requires a confirmation target", self.nodes[2].sendtoaddress, address, 1.0, None, None, None, None, None, 'bTc/kB')
+        txid = self.nodes[2].sendtoaddress(
+            address=address,
+            amount=1.0,
+            conf_target=0.00002500,
+            estimate_mode='btc/kb',
+        )
+        tx_size = self.get_vsize(self.nodes[2].gettransaction(txid)['hex'])
+        self.sync_all(self.nodes[0:3])
+        self.nodes[0].generate(1)
+        self.sync_all(self.nodes[0:3])
+        postbalance = self.nodes[2].getbalance()
+        fee = prebalance - postbalance - Decimal('1')
+        assert_fee_amount(fee, tx_size, Decimal('0.00002500'))
+
+        # send with explicit sat/b fee
+        self.sync_all(self.nodes[0:3])
+        self.log.info("test explicit fee (sendtoaddress as sat/b)")
+        self.nodes[0].generate(1)
+        prebalance = self.nodes[2].getbalance()
+        assert prebalance > 2
+        address = self.nodes[1].getnewaddress()
+        # Throw if no conf_target provided
+        assert_raises_rpc_error(-8, "Selected estimate_mode requires a confirmation target", self.nodes[2].sendtoaddress, address, 1.0, None, None, None, None, None, 'SAT/b')
+        txid = self.nodes[2].sendtoaddress(
+            address=address,
+            amount=1.0,
+            conf_target=2,
+            estimate_mode='SAT/B',
+        )
+        tx_size = self.get_vsize(self.nodes[2].gettransaction(txid)['hex'])
+        self.sync_all(self.nodes[0:3])
+        self.nodes[0].generate(1)
+        self.sync_all(self.nodes[0:3])
+        postbalance = self.nodes[2].getbalance()
+        fee = prebalance - postbalance - Decimal('1')
+        assert_fee_amount(fee, tx_size, Decimal('0.00002000'))
+
```

Comment:
Add a test for negative/too low feerates?

Reply:
Done